src/operations.rs

```diff
@@ -357,6 +364,49 @@ fn default_transaction_options() -> TransactionOptions {
     TransactionOptions::new().transaction_timeout(DEFAULT_TRANSACTION_TIMEOUT)
 }
 
+enum FileResource {
+    Remote(String),
+    Local(PathBuf)
+}
+
+impl FileResource {
+    fn parse(context: &mut ConsoleContext, string: &str) -> Self {
+        if string.starts_with("http://") || string.starts_with("https://") {
+            Self::Remote(string.to_owned())
+        } else {
+            Self::Local(context.convert_path(&string))
+        }
+    }
+
+    fn read_to_string(&self) -> Result<String, Box<dyn Error + Send>> {
+        match self {
+            FileResource::Remote(url) => {
+                let response = ureq::get(url)
+                    .call()
+                    .map_err(|e| Box::new(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        format!("Failed to fetch file from {}: {}", url, e)
+                    )) as Box<dyn Error + Send>)?;
+                response.into_string()
+                    .map_err(|e| Box::new(std::io::Error::new(
+                        std::io::ErrorKind::Other,
+                        format!("Failed to read file content from {}: {}", url, e)
+                    )) as Box<dyn Error + Send>)
```

Comment:
Just to be extra paranoid: I'm wondering whether we should be a bit more careful when downloading files from the internet. Most systems will provide a checksum to verify the authenticity of the download. Maybe we should support a checksum verification and let the user choose whether or not to use it.

The risks aren't huge, given this is a query which will be run against a user-specified database and TypeQL doesn't (yet) have any features that escape the data-model (like `SELECT INTO OUTFILE`) but if we ever add such features, this could turn into a risk.

Reply:
Implemented